GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -127,27 +166,41 @@ subroutine gx_minimax_grid(num_points, e_min, e_max, &
 
     ! Compute the actual weights used for the inhomogeneous cosine/ FT and check whether
     ! the two matrices for the forward/backward transform are the inverse of each other.
-    do j_point = 1, num_points
-       do i_point = 1, num_points
-          cosft_wt(j_point, i_point) = cosft_wt(j_point, i_point)*cos(tau_points(i_point)*omega_points(j_point))
-          cosft_tw(i_point, j_point) = cosft_tw(i_point, j_point)*cos(tau_points(i_point)*omega_points(j_point))
-          sinft_wt(j_point, i_point) = sinft_wt(j_point, i_point)*sin(tau_points(i_point)*omega_points(j_point))
-       end do
-    end do
+    if(.not.my_bare_cossine_weights) then
+      do j_point = 1, num_points
+         do i_point = 1, num_points
+            cosft_wt(j_point, i_point) = cosft_wt(j_point, i_point)*cos(tau_points(i_point)*omega_points(j_point))
+            cosft_tw(i_point, j_point) = cosft_tw(i_point, j_point)*cos(tau_points(i_point)*omega_points(j_point))
+            sinft_wt(j_point, i_point) = sinft_wt(j_point, i_point)*sin(tau_points(i_point)*omega_points(j_point))
+         end do
+      end do
+    else
+      do j_point = 1, num_points
+         do i_point = 1, num_points
+            tmp_cosft_wt(j_point, i_point) = cosft_wt(j_point, i_point)*cos(tau_points(i_point)*omega_points(j_point))
+            tmp_cosft_tw(i_point, j_point) = cosft_tw(i_point, j_point)*cos(tau_points(i_point)*omega_points(j_point))
+         end do
+      end do
+    end if
 
     allocate (mat(num_points, num_points))
-    mat = matmul(cosft_wt, cosft_tw)
+    if(.not.my_bare_cossine_weights) then
+      mat(:,:) = matmul(cosft_wt, cosft_tw)
+    else
+      mat(:,:) = matmul(tmp_cosft_wt, tmp_cosft_tw)
+    endif
```

Comment:
I don't understand what's going on here. Lines 172 and 174 are identical to lines 180 and 181.

And the matrix multiplication done on lines 188 and 190 are identical. So why are temporary arrays or branching logic required? This looks suspect to me.

Reply:
Also, why _introduce_ array slices when they were not there, and are not needed?

Reply:
This is actually important. Read about [-Wrealloc-lhs](https://gcc.gnu.org/onlinedocs/gfortran/Error-and-Warning-Options.html#Error-and-Warning-Options), for instance:

```
If the array is already allocated with the correct shape, consider using a whole-array array-spec (e.g. (:,:,:)) for the variable on the left-hand side to prevent the reallocation check.
```

This has an impact on performance.

Reply:
W.r.t. [-Wrealloc-lhs](https://gcc.gnu.org/onlinedocs/gfortran/Error-and-Warning-Options.html#Error-and-Warning-Options), this is new to me. I'll check it out. 

But you've not responded to the important part of the question. Specifically, sinft_wt will not be assigned in one of the branching logic, and the code for my_bare_cossine_weights or not, looks **identical**. At least, one can scrap the tmp arrays. At worst, it's wrong.